### ISSUE TITLE
PR Request for CSS Box Model Level 3

### ISSUE THREAD

**Issue by fantasai** [w3c/transitions#469]
# Document title, URLs, estimated publication date
https://www.w3.org/TR/css-box-3/
https://drafts.csswg.org/css-box-3/
At your convenience

# Abstract
https://www.w3.org/TR/css-box-3/#abstract

# Status
CR

# Will new features be allowed to be incorporated in the Recommendation?
No

# Link to group's decision to request transition
https://lists.w3.org/Archives/Public/www-style/2022Oct/0022.html

# Changes
Only editorial (mostly markup)

# Requirements satisfied
yes

# Dependencies met (or not)
CSS2, CSS Writing Modes L3

# Wide Review
https://github.com/w3c/transitions/issues/300

# Issues addressed
None

# Formal Objections
None

# Implementation
All browser engines, as proven in the CSS2 and CSS Writing Modes L3 test suites.

# Patent disclosures
None

**Comment by swickr** [comment]
What is the resolution of the 'at risk' features?  They appear to be retained; what tests demonstrate they have been implemented as specified?

**Comment by fantasai** [comment]
@swickr Hm, we're in an interesting bit of dependency here:

- The ruby spec, which is less mature, defines some new types of boxes.
- Like internal table boxes, and unlike most CSS boxes, some of these new ruby layout boxes _don't_ have margins or padding.
- The margin/padding property definitions were updated to specifically exclude these boxes.

We have two choices here:
- Keep the spec as-is
- Remove any mention of ruby layout boxes from the spec to remove the dependency; but then since margin/padding are defined to “all” boxes, there would be an implication that they apply to these particular ruby boxes.

I suspect the former would be more helpful to developers; the latter would remove any dependency on the Ruby Module. What would you prefer we do here?

**Comment by swickr** [comment]
I understand that the interpretation of "at risk" here is not whether the properties _do_ apply but whether they _should_ apply once ruby containers are fully specified.

If the intent of the spec is that the properties should be ignored when applied to ruby containers then there could be tests that demonstrate which implementations do (or do not) ignore.

If this interpretation is correct then I would be comfortable leaving the "applies to ... except" language in the spec with tests that show what implementations are actually doing.  And consider adding an editorial note where the properties are defined stating that the behavior w.r.t ruby containers may change in the future.

We found [propdef-margin](https://drafts.csswg.org/css-box-3/#propdef-margin) that look applicable. 

In any case, the "at risk" language in SoTD is no longer appropriate at Proposed Recommendation stage.


Separately, the Proposed Rec seems to want to refer to several Working Drafts.  Other than Ruby, what risks to CSS Box Model Level 3 spec are there if those dependent specs change?

**Comment by fantasai** [comment]
@swickr Here's a [testcase](https://www.software.hixie.ch/utilities/js/live-dom-viewer/?%3C!DOCTYPE%20html%3E%0A%3Cstyle%3Erbc%20%7B%20display%3A%20ruby-base-container%20%7D%3C%2Fstyle%3E%0A%3Cp%3EThese%20orange%20annotated%20phrase%20should%20look%20identical%20to%20the%20blue%20one%3A%0A%3Cruby%20style%3D%22color%3A%20orange%22%3E%3Crbc%20style%3D%22margin%3A%202em%3B%20padding%3A%202em%22%3Ebase%3C%2Frbc%3E%3Crtc%20style%3D%22margin%3A%202em%3B%20padding%3A%202em%22%3Etext%3C%2Frtc%3E%3C%2Fruby%3E%20%3Cruby%20style%3D%22color%3A%20blue%22%3Ebase%3Crtc%3Etext%3C%2Frtc%3E%3C%2Fruby%3E). Note that it only makes sense in Firefox, because the other browsers don't support ruby text containers / ruby base containers at all.

So I think we've got two options here:

- commit the testcase to WPT, and add a note that behavior might change
- remove the normative statement about ruby boxes, add a note stating that behavior for ruby layout is undefined in this level

Maybe the second is cleaner?

Wrt references, all of them can be downgraded as necessary. They're mostly just pulling terminology from the latest version.

**Comment by plehegar** [comment]
The second approach is cleaner indeed.

Assuming the normative text are removed, respective note(s) is/are added, and the SOTD 'at-risk' paragraph is also removed, the transition is approved.

**Comment by svgeesus** [comment]
@fantasai could you summarize where we are on this. Is it ready to be published now?

**Comment by fantasai** [comment]
@svgeesus Edits made, over to you. :)

**Comment by svgeesus** [comment]
Publication requested 14 Feb expected 16 Feb

**Comment by svgeesus** [comment]
[Published 16 Feb 2023](https://www.w3.org/TR/2023/PR-css-box-3-20230216/)
